### ISSUE TITLE
Add kustomize based deployment

### ISSUE THREAD

**Issue by garyedwards** [eclipse/hono#2305]
Hi thanks for a great project.  Just wondering if there are any kustomize based deployments out there and if so should they be added to this repo as a base?

We are working on something at the moment but would be great to know if there are any existing efforts.

**Comment by sophokles73** [comment]
@garyedwards sorry for the delay and thanks for the question.

I am not aware of anybody having used kustomize for deploying Hono yet. I also have to admit that I am not familiar with kustomize, in fact, I wasn't even aware of its existence until this issue ;-)

Let me take a look at kustomize in order to get a better understanding first before coming back to you with an answer to your question.

**Comment by sophokles73** [comment]
@garyedwards I have taken a look at kustomize. My understanding is that it would need some *base* kubernetes resource descriptors for Hono's components which could then be *kustomize*d, right?
We are providing a Helm chart for Hono which is hosted by the [Eclipse IoT Packages]() project. Therefore, we do not maintain kubernetes resource descriptors as part of Hono (anymore). You should be able to generate a *default*/*base* set of descriptors simply by running `helm template` with the chart's default values and use them as input to any kustomizations.
But maybe you had something different in mind?

**Comment by garyedwards** [comment]
Hi @sophokles73 thanks a lot for looking into this.  Your research is correct you would have a base config and then patches or overlays that customise the base.  I have been using the `helm template` technique as you suggest with some clean up from the files created and it works fine.  A standard base may be nice for the project as kustomize is gaining traction in the CD / GitOpt space.  I understand that it may add a duplication of effort with the Helm Charts so may not desired.

**Comment by sophokles73** [comment]
As you pointed out correctly, such base config files would basically be a duplication of the effort required for maintaining the Helm chart. Personally, I will not have the time to do that and I strongly doubt that other committers will find the time. However, we also have a [Hono Extras](https://github.com/eclipse/hono-extras) repository for community contributions that should not go into Hono's code base. So, if you are looking for a (central) Hono related place to put such files, I can imagine putting them there. Remember, though, that you will probably be responsible for them yourself, i.e. we will not make sure that they are kept in sync with Hono and its Helm chart unless you do. Would that be helpful?

**Comment by garyedwards** [comment]
I would probably just end up automating pulling any helm chart changes using the `helm template` command to keep in sync with changes made by the main developers.  Maybe it is better I work on something for the wiki to highlight the `helm template` workflow for kustomize?  Happy for this to be closed.

**Comment by sophokles73** [comment]
> Maybe it is better I work on something for the wiki to highlight the helm template workflow for kustomize?

Yes, that would be cool. I guess we could [add a section to the documentation](https://www.eclipse.org/hono/docs/dev/deployment/). That would need to go into the `site/documentation/content/deployment` folder ...
